### PR TITLE
Lastupdated filter

### PR DESCRIPTION
**Pull request by lincmba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Resolves [link to issue]

**Engineer Checklist**

- [ ] I have written **Unit tests** for any new feature(s) and edge cases for
      bug fixes
- [ ] I have added documentation for any new feature(s) and configuration
      option(s) on the `README.md`
- [ ] I have run `mvn spotless:check` to check my code follows the project's
      style guide
- [ ] I have run `mvn clean test jacoco:report` to confirm the coverage report
      was generated at `plugins/target/site/jacoco/index.html`
- [ ] I ran `mvn clean package` right before creating this pull request.

### CODE REVIEW

File: plugins/src/test/java/org/smartregister/fhir/gateway/plugins/UtilsTest.java [hunk 1]
```diff
@@ -179,4 +188,77 @@ public void testReadApplicationConfigBinaryResourceWithEmptyResourceId() {
...
+        List<Bundle.BundleLinkComponent> links = resultBundle.getLink();
+
+        Bundle.BundleLinkComponent nextLink =
+                links.stream()
+                        .filter(link -> Bundle.LINK_NEXT.equals(link.getRelation()))
+                        .findFirst()
+                        .orElse(null);
+        Assert.assertNotNull(nextLink);
+        // TODO the assertion below should pass
+        // Assert.assertEquals("http://new-base-url/nextPage?param=value", nextLink.getUrl());
```
Comment (lincmba) [inline plugins/src/test/java/org/smartregister/fhir/gateway/plugins/UtilsTest.java:254]: The `cleanBaseUrl` function does not consider the path elements when replacing the base url. Is this an expected behavior or should it add them too. For example: `http://old-base-url/nextPage?param=value` should become `http://new-base-url/nextPage?param=value` instead, it becomes `http://new-base-url?param=value`

Comment (ndegwamartin) [reply plugins/src/test/java/org/smartregister/fhir/gateway/plugins/UtilsTest.java:257]: Good catch , though as mentioned the updates for that I will be making on the original PR here - https://github.com/onaio/fhir-gateway-extension/pull/92

Comment (ndegwamartin) [reply plugins/src/test/java/org/smartregister/fhir/gateway/plugins/UtilsTest.java:257]: Also as recommended you can make the PR to point to main instead
